source/heating_model/tidal_heating.cc

```diff
@@ -0,0 +1,127 @@
+/*
+  Copyright (C) 2011 - 2024 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file LICENSE.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#include <aspect/heating_model/tidal_heating.h>
+
+
+namespace aspect
+{
+  namespace HeatingModel
+  {
+    template <int dim>
+    void
+    TidalHeating<dim>::
+    evaluate (const MaterialModel::MaterialModelInputs<dim> &/*material_model_inputs*/,
+              const MaterialModel::MaterialModelOutputs<dim> &material_model_outputs,
+              HeatingModel::HeatingModelOutputs &heating_model_outputs) const
+    {
+      /*
+        H equation is from Tobie et al. (2003) (https://doi.org/10.1029/2003JE002099)
+        H= 2*(viscosity)*(time-averaged tidal strain rate)^2/(1+((viscosity)*(tidal frequency)/(elastic shear modulus))^2))
+        viscosity = material_model_outputs.viscosities
+        time-averaged strain rate = tidal_strain_rate
+        tidal frequency = tidal_frequency
+        elastic shear modulus = elastic_shear_modulus
+      */
+      for (unsigned int q=0; q<heating_model_outputs.heating_source_terms.size(); ++q)
+        {
+          // return a value varying by viscosity
+          heating_model_outputs.heating_source_terms[q] = 2 * material_model_outputs.viscosities[q] * tidal_strain_rate * tidal_strain_rate
+                                                          / ( 1 + ( (tidal_frequency * material_model_outputs.viscosities[q]) / elastic_shear_modulus ) * ( (tidal_frequency * material_model_outputs.viscosities[q]) / elastic_shear_modulus ) );
+          heating_model_outputs.lhs_latent_heat_terms[q] = 0.0;
+        }
+    }
+
+
+    template <int dim>
+    MaterialModel::MaterialProperties::Property
+    TidalHeating<dim>::
+    get_required_properties () const
+    {
+      return MaterialModel::MaterialProperties::none;
```

Comment:
This is not correct, you need the viscosity to compute the heating term. So the correct term is:
```suggestion
      return MaterialModel::MaterialProperties::viscosity;
```